File: ngoding 1/buTatiCode/python/jualBarang.py
inputKodeBarang = str(input('masukan kode barang :'))
jumlahBeliBarang = int(input('berapa jumlah yang anda beli :'))
bayarSaldo = int(input('masukan saldo anda :'))
# adaDiskon = str('apakah ada diskon ? [YA/TIDAK] :')

tabelBarang = {
    "PK01": {"nama": "Pakaian", "harga_satuan": 75000},
    "TS02": {"nama": "Tas", "harga_satuan": 65500},
    "SP03": {"nama": "Sepatu", "harga_satuan": 157000},
    "AK04": {"nama": "Aksesoris", "harga_satuan": 45000}
}   

if inputKodeBarang not in tabelBarang:
       print('gaada di dalam tabel')
else:
       namaBarang = tabelBarang[inputKodeBarang]["nama"]
       hargaSatuanBarang = tabelBarang[inputKodeBarang]["harga_satuan"]
       namaBarang = tabelBarang[inputKodeBarang]["nama"]
       hargaSatuanBarang = tabelBarang[inputKodeBarang]["harga_satuan"]
       hargaTotal = hargaSatuanBarang * jumlahBeliBarang
       diskonBarang = str(input('apakah di beri diskon ? [YA/TIDAK] :'))
       diskon = 0
       totalBayar = hargaTotal
       kembalian = totalBayar - bayarSaldo
       if diskonBarang == 'YA':
              diberiDiskon = int(input('masukan besaran diskon (%): '))
              diskon = hargaSatuanBarang * (diberiDiskon / 100)
              totalBayar = hargaTotal - diskon
              kembalian = totalBayar - bayarSaldo
  

              
print('_________________________________________________')              
print(f"Kode Barang    : {inputKodeBarang}")
print(f"Nama Barang    : {namaBarang}")
print(f"Jumlah Beli    : {jumlahBeliBarang}")
print(f"Harga Satuan   : Rp {hargaSatuanBarang:,}")
print(f"Diskon         : Rp {diskon:,}%")
print(f"Harga Total    : Rp {hargaTotal:,}")
print(f"Total Bayar    : Rp {totalBayar:,}")
print(f"Bayar          : Rp {bayarSaldo:,}")
print(f"Kembalian      : Rp {kembalian:,}")
            
       
       

File: ngoding 1/pakTaryanaCode/javascript_9/js/jualBarang.js
const proses = () => {
       const kode_barang = document.form.kode_barang.value.toUppercase()
       const jumlah_beli_barang = document.form.jumlah_beli_barang.value
       const bayar_saldo = document.form.bayar_saldo.value

       if (kode_barang === 'PKO1') {
              nama_barang = 'pakaian'
              harga_barang = 70000
       } else if (kode_barang === 'TS02') {
              nama_barang = 'tas'
              harga_barang = 80000
       } else if (kode_barang === 'SP03') {
              nama_barang = 'sepatu'
              harga_barang = 90000
       } else if (kode_barang == 'AK04') {
              nama_barang = 'aksesoris'
              harga_barang = 100000
       } else {
              alert(`tidak ada barang seperti ${kode_barang}`)
       }

       
}